src/Context/CachedContextFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Roave\BetterReflection\Context;
+
+use Roave\BetterReflection\Context\ContextFactory;
+
+class CachedContextFactory implements ContextFactory
+{
+    private $contexts = [];
+
+    public function __construct(ContextFactory $innerFactory)
+    {
+        $this->innerFactory = $innerFactory;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function createForNamespace($namespace, $fileContents)
+    {
+        $hash = $namespace . md5($fileContents);
```

Comment:
Collision prevention. I'd be fine with having `$fileContents` as key (even better)